website/content/docs/auth/kubernetes.mdx

```diff
@@ -101,9 +106,91 @@ management tool.
+injected service account tokens to a year to help smooth the transition to
+short-lived tokens. If you would like to disable this, set
+[--service-account-extend-token-expiration=false][k8s-extended-tokens] for
+`kube-apiserver` or specify your own `serviceAccountToken` volume mount at
+`/var/run/secrets/kubernetes.io/serviceaccount` in your pods. See
+[here](/docs/auth/jwt/oidc_providers#specifying-ttl-and-audience) for an example.
+
+[k8s-1.21-changelog]: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.21.md#api-change-2
+[k8s-extended-tokens]: https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/#options
+
+#### Use the Vault client's JWT as the reviewer JWT
+
+When configuring Kubernetes auth, you can omit the `token_reviewer_jwt`, and Vault
+will use the Vault client's JWT as its own auth token when communicating with
+the Kubernetes TokenReview API. If Vault is running in Kubernetes, you also need
+to set `disable_local_ca_jwt=true`.
+
+This means Vault does not store any JWTs and allows you to use short-lived tokens
+everywhere but adds some operational overhead to maintain the cluster role
+bindings on the set of service accounts you want to be able to authenticate with
+Vault. Each client of Vault would need the `system:auth-delegator` ClusterRole:
+
+```bash
+kubectl create clusterrolebinding vault-client-auth-delegator \
+  --clusterrole=system:auth-delegator \
+  --group=group1 \
+  --serviceaccount=default:svcaccount1 \
+  ...
+```
+
+#### Continue using long-lived tokens
+
+The default Kubernetes secret created for a service account is still long lived,
+and can be used as the `token_reviewer_jwt` without needing to refresh it. To
+find the secret, run:
+
+```bash
+kubectl get secret "$(kubectl get serviceaccount default -o jsonpath='{.secrets[0].name}')"
+```
+
+Using this maintains previous workflows but does not fully take advantage of the
+new default short-lived tokens.
+
+#### Use JWT auth
+
+Kubernetes auth is specialized to use Kubernetes' TokenReview API. However, the
+JWT tokens Kubernetes generates can also be verified using Kubernetes as an OIDC
+provider. The JWT auth method documentation has [instructions][k8s-jwt-auth] for
+setting up JWT auth with Kubernetes as the OIDC provider.
+
+[k8s-jwt-auth]: /docs/auth/jwt/oidc_providers#kubernetes
+
+This solution allows you to use short-lived tokens for all clients and removes
+the need for a reviewer JWT. However, the client tokens cannot be revoked before
+their TTL expires, so it is recommended to keep the TTL short with that
+limitation in mind.
+
 ### Discovering the service account `issuer`
 
--> **Deprecated:** The `issuer` parameter has been deprecated as of Vault 1.9 and will be removed in a future release.
+-> **Note:** From Vault 1.9, `disable_iss_validation` and `issuer` are deprecated
+and the default for `disable_iss_validation` has changed to `true`. The following
```

Comment:
This only applies on new mounts/configs. Should we call out the need to update the config to set this to true if they update their kubernetes clusters to 1.21 and above?

Reply:
Good point. I've added some comments about this.